File: 7-kyu/digits-average/python/solution.py
from math import ceil

def digits_average(input):
    if input < 10:
        return input
    ret = 0
    inp_str = str(input)
    for i in range(len(inp_str) - 1):
        avg = ceil((int(inp_str[i]) + int(inp_str[i + 1])) / 2)
        ret = ret * 10 + avg
    return digits_average(ret)

File: 7-kyu/digits-average/javascript/solution.js
const digitsAverage = input => {
  if (input < 10) {
    return input;
  }
  const digits = input.toString().split("").map(Number);
  const next = [];
  for (let i = 0; i < digits.length - 1; i++) {
    const pair = [digits[i], digits[i + 1]];
    const sum = pair.reduce((acc, nxt) => acc + nxt, 0);
    const avg = Math.ceil(sum / 2);
    next.push(avg);
  }
  const nextNum = next.reduce((acc, nxt) => acc * 10 + nxt, 0);
  return digitsAverage(nextNum);
};
